chain/network/src/config.rs

```diff
@@ -446,7 +447,24 @@ impl NetworkConfig {
         self.routing_table_update_rate_limit
             .validate()
             .context("routing_table_update_rate_limit")?;
-        Ok(VerifiedConfig { node_id: self.node_id(), inner: self })
+
+        // Verify ip address can be signed and verified
+        if !(self.node_addr.is_some()) {
```

Comment:
It seems like this changes the definition of a valid config. Have you thought about the implications?

Reply:
Valid config is simply an internal wrapper around NetworkConfig. It doesnt change how the NetworkConfig is constructed. This passes all existing unit tests, i dont think it affects anything.

Reply:
I don't think passing unit tests is sufficient justification. Did you come across any background information on why `node_addr` is an optional field to begin?

Reply:
Good point, looks like its added here to make the listening address configurable for testing purposes
- https://github.com/near/nearcore/pull/8431

Regardless, the listening address is unreliable as the actual communication happens based on the tcp address instead of the listening address. For instance, in pytest's localnet, the listening address is configured as 0.0.0.0, but the actual tcp connecting refers to 127.0.0.1 

updated to this code to verify it passes the pytests
- https://github.com/near/nearcore/commit/16072e5842478de1293e2a7263db04dd055f8d90

will send another commit to clean up and remove any reference to the listening address for signing ip address purposes